### PR TITLE
Added Support for Python 3 and Django 1.10

### LINKED ISSUE #5: Add support for Python 3.5+
Update the package to support Python 3.5+.

### PR DESCRIPTION
**Pull request by clintonb**
Fixes #5

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce @douglashall please review

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
+  - pip install -U pip wheel codecov
+  - pip install -r requirements/test.txt
+
 script:
-- make validate
+  - make test
+
 after_success:
-- codecov
+  codecov
```
Comment (rlucioni) [inline .travis.yml]: I'm surprised you don't need a hyphen here.

Comment (clintonb) [inline .travis.yml]: YAML is weird. Fixing, nevertheless.

File: setup.py [hunk 2]
```diff
@@ -29,6 +34,6 @@
     license='AGPL',
     packages=find_packages(exclude=['tests', '*.tests']),
     install_requires=[
-        'django>=1.8,<1.9',
+        'django>=1.8,<1.10',
```
Comment (rlucioni) [inline setup.py]: If you're adding support for 1.10 here, shouldn't this read:
```
'django>=1.8,<1.11',
```

**Comment by douglashall** [comment]
This looks good to me. You'll want to remove [references](https://github.com/edx/ecommerce/search?utf8=%E2%9C%93&q=CurrentSiteWithDefaultMiddleware) to `CurrentSiteWithDefaultMiddleware` when upgrading ecommerce to the latest version.

**Comment by clintonb** [comment]
@douglashall please clarify. Is that middleware still needed? If not, why hasn't it already been removed?

File: setup.py (hunk 2 above)
Comment (clintonb) [inline setup.py]: Fixed.

**Comment by douglashall** [comment]
`CurrentSiteWithDefaultMiddleware` was removed with version 2.0.0 of this package. There has not been a compelling reason to upgrade ecommerce up to this point.

**Pull request merged**